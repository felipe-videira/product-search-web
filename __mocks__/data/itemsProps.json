{
    "search": "xbox",
    "items": [
        {
            "id": "MLA865203548",
            "title": "Microsoft Xbox One S 1tb Two-controller Bundle Blanco",
            "price": {
                "currency": "ARS",
                "amount": 55998,
                "decimals": 0
            },
            "picture": "../../__mocks__/data/product_img.jpg",
            "condition": "new",
            "free_shipping": true,
            "city_name": "Villa del Parque"
        },
        {
            "id": "MLA856073032",
            "title": "Microsoft Xbox One S 1tb Pro Evolution Soccer 2019 Bundle Edition Blanco",
            "price": {
                "currency": "ARS",
                "amount": 47900,
                "decimals": 0
            },
            "picture": "../../__mocks__/data/product_img.jpg",
            "condition": "new",
            "free_shipping": true,
            "city_name": "Pilar"
        },
        {
            "id": "MLA870479844",
            "title": "Microsoft Xbox One S 1tb Fortnite Blanco",
            "price": {
                "currency": "ARS",
                "amount": 68990,
                "decimals": 0
            },
            "picture": "../../__mocks__/data/product_img.jpg",
            "condition": "new",
            "free_shipping": true,
            "city_name": "Pilar"
        },
        {
            "id": "MLA875073601",
            "title": "Microsoft Xbox One S 1tb Pro Evolution Soccer 2020 Blanco",
            "price": {
                "currency": "ARS",
                "amount": 48999,
                "decimals": 0
            },
            "picture": "../../__mocks__/data/product_img.jpg",
            "condition": "new",
            "free_shipping": true,
            "city_name": "Caballito"
        }
    ],
    "categories": [
        "Consolas y Videojuegos",
        "Consolas"
    ]
}