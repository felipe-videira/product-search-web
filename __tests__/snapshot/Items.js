import React from 'react';
import renderer from 'react-test-renderer';
import Items from '../../pages/items';
import itemsProps from '../../__mocks__/data/itemsProps.json';

describe('Items', () => {
  it('Should render without crashing', () => {
    const tree = renderer.create(<Items {...itemsProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
