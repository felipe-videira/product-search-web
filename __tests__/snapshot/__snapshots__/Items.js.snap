// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Items Should render without crashing 1`] = `
<div>
  <nav>
    <a
      href="/"
    >
      <img
        alt="Mercado Libre"
        height="50"
        src="/assets/images/logo.png"
        width="75"
      />
    </a>
    <form
      action="/items"
      method="GET"
      name="searchBar"
      onSubmit={[Function]}
    >
      <input
        name="search"
        onChange={[Function]}
        placeholder="Nunca dejes de buscar"
        type="text"
        value="xbox"
      />
      <button
        type="submit"
      >
        <img
          alt="Busca"
          height="20"
          src="/assets/images/search.svg"
          width="20"
        />
      </button>
    </form>
  </nav>
  <main>
    <ul>
      <li>
        Consolas y Videojuegos
      </li>
      <li>
        Consolas
      </li>
    </ul>
    <div>
      <div
        onClick={[Function]}
        onKeyPress={[Function]}
        role="link"
        tabIndex="0"
      >
        <div
          style={
            Object {
              "backgroundImage": "url(../../__mocks__/data/product_img.jpg)",
            }
          }
        />
        <div>
          <div>
            <span>
              $ 55,998
            </span>
            <img
              alt="Flete gratis"
              height="25"
              src="/assets/images/truck.svg"
              width="25"
            />
          </div>
          <h2>
            Microsoft Xbox One S 1tb Two-controller Bundle Blanco
          </h2>
        </div>
        <p>
          Villa del Parque
        </p>
      </div>
      <div
        onClick={[Function]}
        onKeyPress={[Function]}
        role="link"
        tabIndex="0"
      >
        <div
          style={
            Object {
              "backgroundImage": "url(../../__mocks__/data/product_img.jpg)",
            }
          }
        />
        <div>
          <div>
            <span>
              $ 47,900
            </span>
            <img
              alt="Flete gratis"
              height="25"
              src="/assets/images/truck.svg"
              width="25"
            />
          </div>
          <h2>
            Microsoft Xbox One S 1tb Pro Evolution Soccer 2019 Bundle Edition Blanco
          </h2>
        </div>
        <p>
          Pilar
        </p>
      </div>
      <div
        onClick={[Function]}
        onKeyPress={[Function]}
        role="link"
        tabIndex="0"
      >
        <div
          style={
            Object {
              "backgroundImage": "url(../../__mocks__/data/product_img.jpg)",
            }
          }
        />
        <div>
          <div>
            <span>
              $ 68,990
            </span>
            <img
              alt="Flete gratis"
              height="25"
              src="/assets/images/truck.svg"
              width="25"
            />
          </div>
          <h2>
            Microsoft Xbox One S 1tb Fortnite Blanco
          </h2>
        </div>
        <p>
          Pilar
        </p>
      </div>
      <div
        onClick={[Function]}
        onKeyPress={[Function]}
        role="link"
        tabIndex="0"
      >
        <div
          style={
            Object {
              "backgroundImage": "url(../../__mocks__/data/product_img.jpg)",
            }
          }
        />
        <div>
          <div>
            <span>
              $ 48,999
            </span>
            <img
              alt="Flete gratis"
              height="25"
              src="/assets/images/truck.svg"
              width="25"
            />
          </div>
          <h2>
            Microsoft Xbox One S 1tb Pro Evolution Soccer 2020 Blanco
          </h2>
        </div>
        <p>
          Caballito
        </p>
      </div>
    </div>
  </main>
</div>
`;
